guides/Getting Started.md

```diff
@@ -0,0 +1,696 @@
+
+One more final thing to mention here: you can trigger an exception to be thrown by using `Friends.Repo.insert!/2`. If a changeset is invalid, you will see an `Ecto.InvalidChangesetError` exception. Here's a quick example of that:
+
+```
+Friends.Repo.insert! Friends.Person.changeset(%Friends.Person{}, %{ first_name: "Ryan" })
+
+** (Ecto.InvalidChangesetError) could not perform insert because changeset is invalid.
+
+* Changeset changes
+
+%{first_name: "Ryan"}
+
+* Changeset params
+
+%{"first_name" => "Ryan"}
+
+* Changeset errors
+
+[last_name: "can't be blank"]
+
+    lib/ecto/repo/schema.ex:111: Ecto.Repo.Schema.insert!/4
+```
+
+This exception shows us the changes from the changeset, and how the changeset is invalid. This can be useful if you want to insert a bunch of data and then have an exception raised if that data does not insert correctly at all.
+
+Now that we've covered inserting data into the database, let's look at how we can pull that data back out.
+
+## Our first queries
+
+Querying a database requries two steps in Ecto. First, we must construct the query and then we must execute that query against the database by passing the query to the repository. Before we do this, let's re-create the database for our app and setup some test data. To re-create the database, we'll run these commands:
+
+```
+mix ecto.drop
+mix ecto.create
+mix ecto.migrate
+```
+
+Then to create the test data, we'll run this in an `iex -S mix` session:
+
+```elixir
+people = [
+  %Friends.Person{first_name: "Ryan", last_name: "Bigg", age: 28},
+  %Friends.Person{first_name: "John", last_name: "Smith", age: 27},
+  %Friends.Person{first_name: "Jane", last_name: "Smith", age: 26},
+]
+
+Enum.each(people, fn (person) -> Friends.Repo.insert(person) end)
+```
+
+This code will create three new people in our database, Ryan, John and Jane. Note here that we could've used a changeset to validate the data going into the database, but the choice was made not to use one.
+
+We'll be querying for these people in this section. Let's jump in!
+
+### Fetching a single record
+
+Let's start off with fetching just one record from our `people` table:
+
+```elixir
+Friends.Person |> Ecto.Query.first
+```
+
+That code will generate an `Ecto.Query`, which will be this:
+
+```
+#Ecto.Query<from p in Friends.Person, order_by: [asc: p.id], limit: 1>
+```
+
+The code between the angle brackets `<...>` here shows the Ecto query which has been constructed. We could construct this query ourselves with almost exactly the same syntax:
+
+```elixir
+require Ecto.Query
+Ecto.Query.from p in Friends.Person, order_by: [asc: p.id], limit: 1
+```
+
+We need to `require Ecto.Query` here so that the module is available for us to use. Then it's a matter of calling the `from` function from `Ecto.Query` and passing in the code from between the angle brackets. As we can see here, `Ecto.Query.first` saves us from having to specify the `order` and `limit` for the query.
```

Comment:
Require is needed to enables macros from a given module. Regular functions can be used freely without this.